src/util.typ

```diff
@@ -385,3 +385,25 @@
   }
   return body
 }
+
+
+#let str-to-number-regex = regex("^(-?\d*\.?\d+)(cm|mm|pt|em|in|%|deg|rad)?$")
```

Comment:
Should we allow scientific notation here? 
`^(-?\d*\.?\d+)([eE]-?\d+)?(cm|mm|pt|em|in|%|deg|rad)?$`